lxd/api_cluster.go

```diff
@@ -3988,6 +3988,15 @@ func clusterGroupPost(d *Daemon, r *http.Request) response.Response {
 			return fmt.Errorf("Name %q already in use", req.Name)
 		}
 
+		usedBy, err := clusterGroupUsedBy(r.Context(), s, tx, name)
```

Comment:
Some of our "used by" functions take a `firstOnly bool` argument, which would be set true in cases like this, so they can use an optimisation of stopping the search as soon as the first user of the group has been found.

Would be good to use that same technique here in `clusterGroupUsedBy`